### DISCUSSION TITLE
choco install / upgrade stopped working

### DISCUSSION THREAD

**Discussion by saas786** [chocolatey/choco#2396] [Q&A] [answered]
### What You Are Seeing?
```
PS C:\Windows\system32> choco upgrade chocolatey
Chocolatey v0.10.15
Upgrading the following packages:
chocolatey
By upgrading you accept licenses for the packages.
Error retrieving packages from source 'https://chocolatey.org/api/v2/':
 The operation has timed out.

chocolatey was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""; Source(s): "https://chocolatey.org/api/v2/"

Chocolatey upgraded 0/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Warnings:
 - chocolatey - chocolatey was not found with the source(s) listed.
 ```

same with install command:

![ConEmu64_5rXBr0vMEZ](https://user-images.githubusercontent.com/1642796/135707186-12136677-f12c-4668-845f-5859e564b664.png)

### What is Expected?
It should upgrade / install packages.

---

choco seems to work but not when installing / upgrading packages.

![ConEmu64_fKpVGOfZry](https://user-images.githubusercontent.com/1642796/135707136-fd59a6d9-5c4f-4e69-a78b-288ea08e8a9c.png)

### Output Log
`choco install chocolateygui --debug --trace --log-file="'C:\temp\GH2350.log'" -y`

System.Net Error: 0 : [23636] Can't retrieve proxy settings for Uri 'https://chocolatey.org/api/v2/'. error code: 12180.

[GH2350.log](https://github.com/chocolatey/choco/files/7271039/GH2350.log)

---
#### Note: 2022-July-25

I am not sure, but I think it was my VPN running behind the scenes, even though I shutdown the VPN app.

**Comment by gep13** [comment]
@saas786 can I ask that you try again? This sounds like a transient issue with the site rather than anything else.

**Comment by saas786** [comment]
@gep13

Tried again, failed.

```
System.Net Information: 0 : [16452] ConnectStream#14434584::ConnectStream(Buffered -1 bytes.)
System.Net Information: 0 : [16452] Associating HttpWebRequest#14920772 with ConnectStream#14434584
System.Net Information: 0 : [16452] Associating HttpWebRequest#14920772 with HttpWebResponse#62802394
System.Net Information: 0 : [16452] ContentLength=-1
System.Net Information: 0 : [19048] ServicePoint#13858403 - Closed as idle.
System.Net Information: 0 : [19048] ServicePoint#13858403 - ConnectionGroup#4032828 - Closed as idle.
System.Net Information: 0 : [19048] ServicePoint#13858403::ReleaseConnectionGroup(S>)
System.Net Information: 0 : [19048] ServicePoint#13858403::ReleaseConnectionGroup, returning(false)
System.Net.Sockets Error: 0 : [16452] Socket#623043::UpdateStatusAfterSocketError() - TimedOut
System.Net.Sockets Error: 0 : [16452] Exception in Socket#623043::Receive - A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```

**Comment by gep13** [comment]
@saas786 is there any sort of proxy in play within your environment?

I am not aware of any current issues with the site.

@Windos @JPRuskin

**Comment by saas786** [comment]
> @saas786 is there any sort of proxy in play within your environment?
> 
> I am not aware of any current issues with the site.
> 
> @Windos @JPRuskin

I don't think so.

As I don't use any kind of proxies.

Choco was working fine, I have installed a fresh windows on I think 28 Aug, and installed dozen of apps via choco.
But couple of weeks ago, when I wanted to install some app, it failed.
So spent a week or so to figure it out, but no luck, so posted this thread.

If you have some troubleshooting idea, I can give it a go, and provide you the results.

**Comment by JPRuskin** [comment]
> I am not aware of any current issues with the site.

Not aware of anything like this, particularly nothing persisting a few weeks back.

I am seeing two quite different errors between the two examples above, though. We did have the issue relatively recently when a user was unknowingly being redirected to a proxy for some files (#2350). If we run a similar test with debug, we can perhaps look at some more details on the failure.

Does specifying the source as a URL help? (e.g. `choco upgrade chocolatey -s https://chocolatey.org/api/v2/`)

**Comment by saas786** [comment]
Doesn't look similar to #2350
As certificate is verified.

```
System.Net Information: 0 : [11872] SecureChannel#50198807 - Remote certificate was verified as valid by the user.
System.Net Information: 0 : [11872] ProcessAuthentication(Protocol=Tls12, Cipher=Aes128 128 bit strength, Hash=Sha256 0 bit strength, Key Exchange=44550 255 bit strength).
System.Net Information: 0 : [11872] Connection#51294256 - Received status line: Version=1.1, StatusCode=200, StatusDescription=OK.
System.Net Information: 0 : [11872] Connection#51294256 - Received headers
{
Connection: close
Vary: Accept-Encoding
DataServiceVersion: 2.0;
X-Frame-Options: deny
CF-Cache-Status: HIT
Age: 13401
Expect-CT: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
Strict-Transport-Security: max-age=12960000
X-Content-Type-Options: nosniff
CF-RAY: 6994ab9e3bb41d39-BLR
Cache-Control: max-age=86400
Content-Type: application/xml;charset=utf-8
Date: Tue, 05 Oct 2021 06:42:30 GMT
Last-Modified: Tue, 05 Oct 2021 02:59:09 GMT
Server: cloudflare
}.
System.Net Information: 0 : [11872] ConnectStream#49136085::ConnectStream(Buffered -1 bytes.)
System.Net Information: 0 : [11872] Associating HttpWebRequest#57895144 with ConnectStream#49136085
System.Net Information: 0 : [11872] Associating HttpWebRequest#57895144 with HttpWebResponse#39571581 System.Net Information: 0 : [11872] ContentLength=-1
System.Net Information: 0 : [6568] ServicePoint#21749735 - Closed as idle.
System.Net Information: 0 : [6568] ServicePoint#21749735 - ConnectionGroup#4032828 - Closed as idle.
System.Net Information: 0 : [6568] ServicePoint#21749735::ReleaseConnectionGroup(S>)
System.Net Information: 0 : [6568] ServicePoint#21749735::ReleaseConnectionGroup, returning(false)
```

**Comment by saas786** [comment]
> Does specifying the source as a URL help? (e.g. `choco upgrade chocolatey -s https://chocolatey.org/api/v2/`)

Nope it didn't:
```
choco upgrade chocolatey -s https://chocolatey.org/api/v2/ --debug --trace --log-file="'C:\temp\GH2389.log'" -y
```
[GH2389.log](https://github.com/chocolatey/choco/files/7283431/GH2389.log)

**Comment by gep13** [comment]
@saas786 I am going to move this over to a Discussion, as I don't believe this is an issue with Chocolatey specifically, but more something in your environment that is causing it not to work.

**Comment by saas786** [comment, marked as answer]
@gep13 No problem, I just want it resolved, so I can continue to use Choco.

**Reply by gep13** [reply]
@saas786 would it be possible for you to try using something other than `Cmder`?  I would like to rule out that causing problems.  Thanks

**Reply by saas786** [reply]
> @saas786 would it be possible for you to try using something other than `Cmder`? I would like to rule out that causing problems. Thanks

It didn't worked with powershell, cmd nor cmder, mintty etc.

Strangely it works now, just tried to install / upgrade packages both works.

**Reply by gep13** [reply]
@saas786 glad to hear that you got this working again.  Let us know again if you start to run into problems.

**Reply by fafg** [reply]
Installing the following packages:
cmake
By installing, you accept licenses for the packages.
[NuGet] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond 104.20.73.28:80
cmake not installed. An error occurred during installation:
 Unable to resolve dependency 'cmake.install (= 3.23.0)'.
cmake package files install failed with exit code 1. Performing other installation steps.
The install of cmake was NOT successful.
cmake not installed. An error occurred during installation:
 Unable to resolve dependency 'cmake.install (= 3.23.0)'.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - cmake (exited 1) - cmake not installed. An error occurred during installation:
 Unable to resolve dependency 'cmake.install (= 3.23.0)'.
choco error
At C:\Users\vsts\install.ps1:87 char:28
+ if ($LASTEXITCODE -eq 1) { throw "choco error" }
+                            ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (choco error:String) [], Runti 
   meException
    + FullyQualifiedErrorId : choco error

**Reply by fafg** [reply]
My setup script can run properly locally, however executing it on my build agent I got this error, in the same place and same package.

**Reply by pauby** [reply]
@fafg This discussion has been answered. Can you create a new discussion? Can you also indicaee in that discussion what you've done to troubleshoot it.

**Comment by jpluimers** [comment]
If this was not a client side issue, it might be a candidate for incorporation in #37 (currently titled "Receive messages (like maintenance messages) from servers").

Please all upvote the first comment there so it gets more traction as the goal is to make the `choco` client be more informative of ongoing maintenance (and hopefully also about outages).

**Reply by smling** [reply]
I got the same problem on cannot run `choco upgrade all -y`, but it status request timeout. Checked with browser with same URL and xml return. So it seems not caused by network connectivity. 

Has any problem in server side???

![image](https://user-images.githubusercontent.com/10277968/180655314-e3d02dcd-dbe0-4746-b13f-63e6db1df5e3.png)

**Reply by TheCakeIsNaOH** [reply]
@smling see https://status.chocolatey.org/ for if any of the Chocolatey websites or services are known to be down. Currently, nothing is showing as down or degraded. 

It could be something on your end, or something unknown on the Community Repository end. The repository for reporting issues with connecting to the Community Repository is this one: https://github.com/chocolatey/home

I'd suggest updating your source to point to `community.chocolatey.org` instead of `chocolatey.org`, but beyond that, looking through the logs of any anti malware and/or firewall programs you have running and see if any of them are blocking requests would be a good place to start.

**Comment by jerviscui** [comment]
I have what looks like the same problem.

I get the logs when I run choco,

> System.Net Information: 0 : [23592] SecureChannel#36470260 - 远程证书经用户验证为有效。
System.Net Information: 0 : [23592] ProcessAuthentication(协议=Tls12，密码=Aes128 128 位强度，哈希=Sha256 0 位强度，密钥交换=44550 255 位强度)。
System.Net Information: 0 : [23592] Connection#33092577 - 收到的状态行: Version=1.1，StatusCode=200，StatusDescription=OK。
System.Net Information: 0 : [23592] Connection#33092577 - 已收到标头
{
Connection: close
vary: Accept-Encoding
dataserviceversion: 2.0;
x-frame-options: deny
CF-Cache-Status: HIT
Age: 10991
Strict-Transport-Security: max-age=12960000
X-Content-Type-Options: nosniff
CF-RAY: 7509e9c52ce6964f-SJC
Cache-Control: max-age=86400
Content-Type: application/xml;charset=utf-8
Date: Mon, 26 Sep 2022 06:23:50 GMT
Last-Modified: Mon, 26 Sep 2022 03:20:39 GMT
Server: cloudflare
}。
System.Net Information: 0 : [23592] ConnectStream#59796890::ConnectStream(已缓冲 -1 个字节。)
System.Net Information: 0 : [23592] Associating HttpWebRequest#11133493 with ConnectStream#59796890
System.Net Information: 0 : [23592] Associating HttpWebRequest#11133493 with HttpWebResponse#1301106
System.Net Information: 0 : [23592] ContentLength=-1
System.Net.Sockets Error: 0 : [23592] Socket#32406629::UpdateStatusAfterSocketError() - TimedOut
System.Net.Sockets Error: 0 : [23592] Socket#32406629::Receive - 由于连接方在一段时间后没有正确答复或连接的主机没有反应 ，连接尝试失败。 中的异常。

I see chocolatey use a proxy connection:

> System.Net.Sockets Information: 0 : [23592] Socket#56174818 - 已创建从 127.0.0.1:55525 到 127.0.0.1:1080 的连接。
System.Net Information: 0 : [23592] Connection#65893970 - 已创建从 127.0.0.1:55525 到 127.0.0.1:1080 的连接。

But I don't set the proxy in the config. The config is default value `key="proxy" value=""`.

**When I turn off the `ShadowsocksR.exe`, the chocolatey will work fine.**

**Reply by jerviscui** [reply]
When I turn off the `ShadowsocksR.exe`, the new trace info is: 

> System.Net Information: 0 : [21492] ConnectStream#33092577::ConnectStream(已缓冲 -1 个字节。)
System.Net Information: 0 : [21492] Associating HttpWebRequest#9915935 with ConnectStream#33092577
System.Net Information: 0 : [21492] Associating HttpWebRequest#9915935 with HttpWebResponse#29397739
System.Net Information: 0 : [21492] ContentLength=-1
System.Net Information: 0 : [21492] Associating HttpWebRequest#24592149 with ServicePoint#20599914
System.Net Information: 0 : [21492] Associating Connection#20002757 with HttpWebRequest#24592149
System.Net.Sockets Information: 0 : [21492] Socket#45807089 - 已创建从 10.99.59.47:63222 到 104.20.73.28:443 的连接。
System.Net Information: 0 : [21492] Connection#20002757 - 已创建从 10.99.59.47:63222 到 104.20.73.28:443 的连接。
System.Net Information: 0 : [21492] TlsStream#19386696::.ctor(host=community.chocolatey.org, #certs=0, checkCertificateRevocationList=False, sslProtocols=Default, Tls11, Tls12)
System.Net Information: 0 : [21492] Associating HttpWebRequest#24592149 with ConnectStream#40262542
System.Net Information: 0 : [21492] HttpWebRequest#24592149 - Request: GET /api/v2/Packages()?$filter=(tolower(Id)%20eq%20'chocolatey')%20and%20IsLatestVersion HTTP/1.1

Is it because of new connection and TlsStream?

**Comment by Drest** [comment]
I know this is an old thread, but I was having this exact issue upgrading from Chocolatey v0.9.9.11 to v2.2.2
I ended up making a local repository with v.1.2.0 in it, which upgraded without error.
After getting v.1.2.0 installed, it started working normally again.
Steps to follow:

Browse the version history for the chocolatey package. Select v1.2.0
https://community.chocolatey.org/packages/chocolatey#versionhistory

On the left side panel, under Package Specific, click the Download button.
This will download chocolatey.1.2.0.nupkg
Put the file in a locally accessible folder. I chose "C:\Downloads" in this example.

Add that folder as a local source:
choco source add -n=local -s="C:\Downloads"

Now run the upgrade locally:
choco upgrade chocolatey --version=1.2.0

Upgrade should finish successfully. Restart your shell.
Now run the upgrade again. It should now connect to the remote community server:
choco upgrade chocolatey

Optionally disable or remove the local source:
choco source disable -n=local
or
choco source remove -n=local

As this is the top hit on major search providers for this error message, I hope others find it helpful.

**Reply by JPRuskin** [reply]
For what it's worth, our [upgrading to v2](https://docs.chocolatey.org/en-us/guides/upgrading-to-chocolatey-v2-v6#latest-1.x-and-5.x-stable-chocolatey-products) page says that "you need to ensure you are at the latest stable version of the Chocolatey products" before upgrading.

**Comment by pauby** [comment]
I've locked this discussion as it has been answered.
